docs/philosophy.md

```diff
@@ -0,0 +1,41 @@
+---
+title: Philosophy
+---
+
+Jekyll offers a unique philosophy when approaching the problem of static
+site generation. This core philosophy drives development and product
+decisions. When a contributor, maintainer, or user asks herself what Jekyll
+is about, the following principles should come to mind:
+
+### 1. No Magic
+
+Jekyll is not magic. A user should be able to understand the underlying
+processes that make up the Jekyll build without much reading. It should
+behave "as you'd expect."
+
+### 2. It "Just Works"
+
+The out-of-the-box experience should be that it "just works." Run `gem
+install jekyll` and it should build any Jekyll site that it's given.
+Features like auto-regeneration and settings like the markdown renderer
+should represent sane defaults that work perfectly for the vast majority of
+cases. The burden of configuration should not be placed on the user.
+
+### 3. Content is King
+
+Why is Jekyll so loved by content creators? It focuses on content first and
+foremost, making the process of publishing content on the Web easy. Users
+should find the management of their content enjoyable and simple.
+
+### 4. Stability
+
+If a user's site builds today, it should build tomorrow.
+Backwards-compatibility should be strongly preferred over breaking changes.
+Upon breaking changes, provide a clear path for users to upgrade.
```

Comment:
Also maybe that when it comes to breaking changes, practicality over purity. There should be a strong reason to institution a breaking change, and it needs to outweigh the burden we're pushing on users. "I would have done it this way", "this way is easier to read", or "this is the textbook approach" aren't compelling reasons to break a user's site.